packages/types/src/index.ts

```diff
@@ -31,20 +31,38 @@ export interface Dialog {
   state?: string;
 }
 
-export interface View {
-  title?: PlainTextElement;
-  type: 'home' | 'modal' | 'workflow_step';
+interface HomeView {
+  type: 'home';
   blocks: (KnownBlock | Block)[];
+  private_metadata?: string;
   callback_id?: string;
+  external_id?: string;
+}
+
+interface ModalView {
+  type: 'modal';
+  title: PlainTextElement;
+  blocks: (KnownBlock | Block)[];
   close?: PlainTextElement;
   submit?: PlainTextElement;
   private_metadata?: string;
+  callback_id?: string;
   clear_on_close?: boolean; // defaults to false
   notify_on_close?: boolean; // defaults to false
-  submit_disabled?: boolean; // defaults to false
   external_id?: string;
+  submit_disabled?: boolean; // defaults to false
```

Comment:
The API document may not be clear enough but as of today, this option is available only for "workflow_step" type modals. Can you remove `submit_disabled` from this type definition? 
```suggestion
```